jax_lensing/datasets/kappatng.py

```diff
@@ -0,0 +1,71 @@
+"""kappatng dataset."""
+
+import os
+import numpy as np
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+# TODO(kappatng): Markdown description  that will appear on the catalog page.
+_DESCRIPTION = """
+Description is **formatted** as markdown.
+
+It should also contain any processing which has been applied (if any),
+(e.g. corrupted example skipped, images cropped,...):
+"""
+
+# TODO(kappatng): BibTeX citation
+_CITATION = """
+"""
+
+
+class Kappatng(tfds.core.GeneratorBasedBuilder):
+  """DatasetBuilder for kappatng dataset."""
+
+  VERSION = tfds.core.Version('1.0.0')
+  RELEASE_NOTES = {
+      '1.0.0': 'Initial release.',
+  }
+
+  def _info(self) -> tfds.core.DatasetInfo:
+    """Returns the dataset metadata."""
+    # TODO(kappatng): Specifies the tfds.core.DatasetInfo object
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            # These are the features of your dataset like images, labels ...
+            "map": tfds.features.Tensor(shape=[1024,1024], dtype=tf.float32),
+	}),
+        # If there's a common (input, target) tuple from the
+        # features, specify them here. They'll be used if
+        # `as_supervised=True` in `builder.as_dataset`.
+        supervised_keys=("map", "map"),
+        homepage='https://dataset-homepage/',
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager: tfds.download.DownloadManager):
+    """Returns SplitGenerators."""
+    # TODO(kappatng): Downloads the data and defines the splits
+    #path = dl_manager.download_and_extract('https://todo-data-url')
+    data_path = dl_manager.extract("/linkhome/rech/genpuc01/utb76xl/commun/kappaTNG/kappaTNGCosmos.tar.gz")
```

Comment:
ok yeah... hum... we should try to find a way to use a command line argument to specify this path instead of harcoding it in the file. You can check out what they do for the celeb_a_hq dataset for instance: https://www.tensorflow.org/datasets/catalog/celeb_a_hq
https://github.com/tensorflow/datasets/blob/58b03f88aa46d2748882e0f91f3276902130251e/tensorflow_datasets/image/celebahq.py#L123

Reply:
@b-remy did you have a look at this? Just let me know if you would like me to have a go at it :-)

Reply:
Not yet, I am going to take a look now!

Reply:
Alright, this works by specifying the `data_path` in the building command line as:
```
tfds build --manual_dir=data_path
```